firmware/application/apps/ble_app.hpp

```diff
@@ -0,0 +1,232 @@
+
+    uint64_t macAddress;
+    int dbValue;
+    BlePacketData packetData;
+
+    BleRecentEntry()
+        : BleRecentEntry{0} {
+    }
+
+    BleRecentEntry(
+        const uint64_t macAddress)
+        : macAddress{macAddress},
+          dbValue{},
+          packetData{} {
+    }
+
+    Key key() const {
+        return macAddress;
+    }
+
+    // void update(const BlePacketData * packet);
+};
+
+using BleRecentEntries = RecentEntries<BleRecentEntry>;
+using BleRecentEntriesView = RecentEntriesView<BleRecentEntries>;
+
+class BleRecentEntryDetailView : public View {
+   public:
+    BleRecentEntryDetailView(NavigationView& nav, const BleRecentEntry& entry);
+
+    void set_entry(const BleRecentEntry& new_entry);
+    const BleRecentEntry& entry() const { return entry_; };
+
+    void update_data();
+    void focus() override;
+    void paint(Painter&) override;
+
+   private:
+    NavigationView& nav_;
+    BleRecentEntry entry_{};
+
+    Labels labels{
+        {{0 * 8, 0 * 16}, "Len", Color::light_grey()},
+        {{5 * 8, 0 * 16}, "Type", Color::light_grey()},
+        {{10 * 8, 0 * 16}, "Value", Color::light_grey()},
+    };
+
+    Button button_done{
+        {125, 224, 96, 24},
+        "Done"};
+
+    bool send_updates{false};
+
+    Rect draw_field(
+        Painter& painter,
+        const Rect& draw_rect,
+        const Style& style,
+        const std::string& label,
+        const std::string& value);
+};
+
+class BLERxView : public View {
+   public:
+    BLERxView(NavigationView& nav);
+    ~BLERxView();
+
+    void set_parent_rect(const Rect new_parent_rect) override;
+    void paint(Painter&) override{};
+
+    void focus() override;
+
+    std::string title() const override { return "BLE RX"; };
+
+   private:
+    void on_data(BlePacketData* packetData);
+
+    NavigationView& nav_;
+    RxRadioState radio_state_{
+        2402000000 /* frequency */,
+        4000000 /* bandwidth */,
+        4000000 /* sampling rate */,
+        ReceiverModel::Mode::WidebandFMAudio};
+    app_settings::SettingsManager settings_{
+        "BLE Rx", app_settings::Mode::RX};
+
+    uint8_t console_color{0};
+    uint32_t prev_value{0};
+    uint8_t channel_number = 37;
+
+    static constexpr auto header_height = 12 + 2 * 16;
+
+    RFAmpField field_rf_amp{
+        {16 * 8, 0 * 16}};
+
+    LNAGainField field_lna{
+        {18 * 8, 0 * 16}};
+
+    VGAGainField field_vga{
+        {21 * 8, 0 * 16}};
+
+    RSSI rssi{
+        {24 * 8, 0, 6 * 8, 4}};
+
+    Channel channel{
+        {24 * 8, 5, 6 * 8, 4}};
+
+    RxFrequencyField field_frequency{
+        {6 * 8, 0 * 16},
+        nav_};
+
+    OptionsField options_region{
+        {0 * 8, 0 * 8},
+        5,
+        {{"Ch.37 ", 0},
+         {"Ch.38", 1},
+         {"Ch.39", 2}}};
+
+    Console console{
+        {0, 4 * 16, 240, 240}};
+
+    Checkbox check_log{
+        {0 * 8, 1 * 16},
+        3,
+        "LOG",
+        false};
+
+    std::string str_log{""};
```

Comment:
Just FYI: If you want to have your logging setting persisted between app instances, you can use use the settings_ instance. Take a look at like pogsag or remote for examples. There's also some neat binder code that takes care of handling binding the UI control to the variable.